packetbeat/protos/http/http.go

```diff
@@ -583,82 +589,42 @@ func parseCookieValue(raw string) string {
 	return raw
 }
 
-func (http *httpPlugin) extractBody(m *message) []byte {
-	body := []byte{}
-
-	if len(m.contentType) > 0 && http.shouldIncludeInBody(m.contentType) {
-		if len(m.chunkedBody) > 0 {
-			body = append(body, m.chunkedBody...)
-		} else {
-			if isDebug {
-				debugf("Body to include: [%s]", m.raw[m.bodyOffset:])
-			}
-			body = append(body, m.raw[m.bodyOffset:]...)
-		}
-	}
-
-	return body
-}
-
-func (http *httpPlugin) cutMessageBody(m *message) []byte {
-	cutMsg := []byte{}
-
-	// add headers always
-	cutMsg = m.raw[:m.bodyOffset]
-
-	// add body
-	return append(cutMsg, http.extractBody(m)...)
-}
-
-func (http *httpPlugin) shouldIncludeInBody(contenttype []byte) bool {
-	includedBodies := http.includeBodyFor
-	for _, include := range includedBodies {
-		if bytes.Contains(contenttype, []byte(include)) {
-			if isDebug {
-				debugf("Should Include Body = true Content-Type %s include_body %s",
-					contenttype, include)
-			}
-			return true
-		}
-		if isDebug {
-			debugf("Should Include Body = false Content-Type %s include_body %s",
-				contenttype, include)
-		}
-	}
-	return false
-}
-
 func (http *httpPlugin) hideHeaders(m *message) {
 	if !m.isRequest || !http.redactAuthorization {
 		return
 	}
 
-	msg := m.raw
+	msg := m.rawHeaders
+	limit := len(msg)
 
 	// byte64 != encryption, so obscure it in headers in case of Basic Authentication
 
 	redactHeaders := []string{"authorization", "proxy-authorization"}
 	authText := []byte("uthorization:") // [aA] case insensitive, also catches Proxy-Authorization:
 
-	authHeaderStartX := m.headerOffset
-	authHeaderEndX := m.bodyOffset
+	authHeaderStartX := bytes.Index(msg, constCRLF)
+	if authHeaderStartX < 0 {
+		return
+	}
+	authHeaderStartX += len(constCRLF)
+	authHeaderEndX := limit
```

Comment:
why remove the `header/bodyOffset` in favour of bytes.Index? They still seem to be usefull when postprocessing messages.

Reply:
Removal of headerOffset was a mistake. bodyOffset is unnecessary as the headers and body are now stored in the different slices.